src/bp/core/repositories/conversations.ts

```diff
@@ -0,0 +1,238 @@
+        .update({ last_heard_on: this.database.knex.date.now() }),
+      () => ({
+        ...message,
+        sent_on: new Date(),
+        message_raw: raw,
+        message_data: data
+      })
+    )
+  }
+
+  async appendBotMessage(
+    botName: string,
+    botAvatar: string,
+    conversationId: string,
+    payload: sdk.conversations.MessagePayload
+  ): Promise<any> {
+    const { type, text, raw, data } = payload
+    const message = {
+      id: uuid.v4(),
+      conversationId: conversationId,
+      userId: undefined,
+      full_name: botName,
+      avatar_url: botAvatar,
+      message_type: type,
+      message_text: text,
+      message_raw: this.database.knex.json.set(raw),
+      message_data: this.database.knex.json.set(data),
+      sent_on: this.database.knex.date.now()
+    }
+
+    await this.database.knex(messageTable).insert(message)
+
+    return Object.assign(message, {
+      sent_on: new Date(),
+      message_raw: this.database.knex.json.get(message.message_raw),
+      message_data: this.database.knex.json.get(message.message_data)
+    })
+  }
+
+  async listConversations(userId: string, botId: string) {
+    const conversations = (await this.database
+      .knex(conversationsTable)
+      .select('id')
+      .where({ userId, botId })
+      .orderBy('last_heard_on', 'desc')
+      .limit(100)
+      .then()) as any[]
+
+    const conversationIds = conversations.map(c => c.id)
+
+    let lastMessages = this.database.knex
+      .from(messageTable)
+      .distinct(this.database.knex.raw('ON ("conversationId") *'))
+      .orderBy('conversationId')
+      .orderBy('sent_on', 'desc')
+
+    if (this.database.knex.isLite) {
+      const lastMessagesDate = this.database
+        .knex(messageTable)
+        .whereIn('conversationId', conversationIds)
+        .groupBy('conversationId')
+        .select(this.database.knex.raw('max(sent_on) as date'))
+
+      lastMessages = this.database.knex
+        .from(messageTable)
+        .select('*')
+        .whereIn('sent_on', lastMessagesDate)
+    }
+
+    return this.database.knex
+      .from(function(this: any) {
+        this.from(conversationsTable)
+          .where({ userId, botId })
+          .as('wc')
+      })
+      .leftJoin(lastMessages.as('wm'), 'wm.conversationId', 'wc.id')
+      .orderBy('wm.sent_on', 'desc')
+      .select(
+        'wc.id',
+        'wc.title',
+        'wc.description',
+        'wc.logo_url',
+        'wc.created_on',
+        'wc.last_heard_on',
+        'wm.message_type',
+        'wm.message_text',
+        this.database.knex.raw('wm.full_name as message_author'),
+        this.database.knex.raw('wm.avatar_url as message_author_avatar'),
+        this.database.knex.raw('wm.sent_on as message_sent_on')
+      )
+  }
+
+  async getConversation(userId: string, conversationId: string, botId: string) {
+    const condition: any = { userId, botId }
+
+    if (conversationId && conversationId !== 'null') {
```

Comment:
that's a funny representation of null, maybe set the parameter as optional in the method?